core/utils/util.c

```diff
@@ -58,7 +58,7 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  * The ID of this federate. For a non-federated execution, this will be -1.
  * For a federated execution, it will be assigned in the generated code.
  */
-int _lf_my_fed_id = -1;
+uint16_t _lf_my_fed_id = 0xffff;
```

Comment:
Can we introduce a macro (e.g. `LF_INVALID_FED_IC`) instead of a magic number?

Reply:
```suggestion
uint16_t _lf_my_fed_id = UINT16_MAX;  // Federate IDs are counted up from 0. If we hit this, we have too many.
```

Reply:
I consider UINT16_MAX also as a magic number. Unlike a specifically named variable, it does not capture the intended use. And you don't ever want to change the name of `UINT16_MAX`, but we might want to change the invalid (or maximum) federation ID.

Reply:
Good point. I suggest a macro like LF_FED_ID_LIMIT to make it clear that it is an upper bound.